packages/cli/src/main/kotlin/elide/tool/cli/cmd/repl/ToolShellCommand.kt

```diff
@@ -1080,17 +1113,49 @@ private typealias ContextAccessor = () -> PolyglotContext
     }
   }
 
+  private fun KotlinJarBundleInfo.asSource(): Source {
+    TODO("Not yet implemented: ${this::class.simpleName}")
+  }
+
   // Resolve a compiler for the provided language, compile the entrypoint, and return the resolved executable symbol.
-  private fun compileEntrypoint(language: GuestLanguage, ctxAccessor: ContextAccessor, entry: File): Value {
-    when (language) {
+  private fun compileEntrypoint(language: GuestLanguage, ctxAccessor: ContextAccessor, entry: File): ShellRunnable {
+    return when (language) {
       // use the host java compiler, which supports up to the maximum JVM bytecode level anyway
-      JAVA -> ToolProvider.getSystemJavaCompiler().let {
-        error("No way to compile pure Java yet")
+      JAVA -> ShellRunnable.HostExecutable {
+        ToolProvider.getSystemJavaCompiler().let {
+          error("No way to compile pure Java yet")
+        }
       }
 
       // use the embedded kotlin compiler to compile to java bytecode first
-      KOTLIN -> {
-        error("No Kotlin compilation support on-the-fly yet")
+      KOTLIN -> ShellRunnable.HostExecutable {
+        val sourceInfo = Precompiler.PrecompileSourceInfo(
+          name = entry.name,
+          path = entry.toPath(),
+        )
+        val config = KotlinCompilerConfig.DEFAULT
+
+        runBlocking(Dispatchers.IO) {
```

Comment:
The dispatcher here is a bit redundant; since it's blocking the calling thread anyway, might as well use it

Reply:
So you'd recommend `Dispatchers.Default` instead, right? Since we're not in a suspending context here... need to pick something. We can probably eventually make this block suspendable

Reply:
Discussed offline

Reply:
On a second look, `Dispatcher.IO` is fine, because it allows anything suspending deeper in that block to use the thread pool. It's probably not really needed since it's only a simple compiler call but it doesn't hurt either.